I've added the tests, for `EDAnalyzer`, `EDProducer`, `EDFilter`, `OutputModule`.

Without the fix, I get the error
```
$ python3 -m FWCore.ParameterSet.Config
......E.............................
======================================================================
ERROR: testFinalPath (__main__.TestModuleCommand)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/user/fwyzard/CMSSW_14_0_0_pre0/src/FWCore/ParameterSet/python/Config.py", line 3360, in testFinalPath
    self.assertRaises(RuntimeError, p.fillProcessDesc, pset)
  File "/data/cmssw/el8_amd64_gcc12/external/python3/3.9.14-4612d00f9f0430a19291545f1e47b4a4/lib/python3.9/unittest/case.py", line 739, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/data/cmssw/el8_amd64_gcc12/external/python3/3.9.14-4612d00f9f0430a19291545f1e47b4a4/lib/python3.9/unittest/case.py", line 201, in handle
    callable_obj(*args, **kwargs)
  File "/data/user/fwyzard/CMSSW_14_0_0_pre0/src/FWCore/ParameterSet/python/Config.py", line 1488, in fillProcessDesc
    self._insertPaths(adaptor, nodeVisitor)
  File "/data/user/fwyzard/CMSSW_14_0_0_pre0/src/FWCore/ParameterSet/python/Config.py", line 1290, in _insertPaths
    names.extend( [p.label_ for p in finalpathValidator.producersOnFinalpaths])
  File "/data/user/fwyzard/CMSSW_14_0_0_pre0/src/FWCore/ParameterSet/python/Config.py", line 1290, in <listcomp>
    names.extend( [p.label_ for p in finalpathValidator.producersOnFinalpaths])
AttributeError: 'str' object has no attribute 'label_'

----------------------------------------------------------------------
Ran 36 tests in 0.150s

FAILED (errors=1)
```

With the fix:
```
$ python3 -m FWCore.ParameterSet.Config
....................................
----------------------------------------------------------------------
Ran 36 tests in 0.148s

OK
```

--- a/FWCore/ParameterSet/python/Config.py
+++ b/FWCore/ParameterSet/python/Config.py
@@ -1285,9 +1285,9 @@ class Process(object):
               iFinalPath.resolve(self.__dict__)
               finalpathValidator.setLabel(finalpathname)
               iFinalPath.visit(finalpathValidator)
-              filtersOrProducers = finalpathValidator.filtersOnFinalpaths + finalpathValidator.producersOnFinalpaths
-              if filtersOrProducers:
-                  raise RuntimeError("FinalPath %s has non OutputModules %s" % (finalpathname, ",".join(filtersOrProducers)))
+              invalidModules = finalpathValidator.invalidModulesOnFinalpaths
+              if invalidModules:
+                  raise RuntimeError("FinalPath %s has non OutputModules %s" % (finalpathname, ",".join(invalidModules)))
               modulesOnFinalPath.extend(iFinalPath.moduleNames())
           for m in modulesOnFinalPath:
             mod = getattr(self, m)
@@ -3351,7 +3351,7 @@ process.s2 = cms.Sequence(process.a+(process.a+process.a))""")
             p.anal = EDAnalyzer("MyAnalyzer")
             p.t = FinalPath(p.anal)
             pset = TestMakePSet()
-            p.fillProcessDesc(pset)
+            self.assertRaises(RuntimeError, p.fillProcessDesc, pset)
 
             p.prod = EDProducer("MyProducer")
             p.t = FinalPath(p.prod)

--- a/FWCore/ParameterSet/python/SequenceVisitors.py
+++ b/FWCore/ParameterSet/python/SequenceVisitors.py
@@ -72,8 +72,7 @@ class FinalPathValidator(object):
     def __init__(self):
         self.__label = ''
         self._levelInTasks = 0
-        self.filtersOnFinalpaths = []
-        self.producersOnFinalpaths = []
+        self.invalidModulesOnFinalpaths = []
     def setLabel(self,label):
         self.__label = "'"+label+"' "
     def enter(self,visitee):
@@ -88,10 +87,8 @@ class FinalPathValidator(object):
             self._levelInTasks += 1
         if self._levelInTasks > 0:
             return
-        if isinstance(visitee,EDFilter):
-            self.filtersOnFinalpaths.append(visitee.type_())
-        if isinstance(visitee,EDProducer):
-            self.producersOnFinalpaths.append(visitee.type_())
+        if isinstance(visitee,(EDAnalyzer,EDProducer,EDFilter)):
+            self.invalidModulesOnFinalpaths.append(visitee.type_())
     def leave(self,visitee):
         if self._levelInTasks > 0:
             if isinstance(visitee, Task):

